### PR TITLE
Ensure that the tutorial is kept up-to-date.

### PR DESCRIPTION
**Pull request by ahmedcharles**

### CODE REVIEW

**Comment by alexcrichton** [comment]
Whoa awesome, thanks for the PR! It looks like there's some CI failures but I think that's just due to depending on git versions instead of crates.io versions perhaps?

Also, I wonder if we could perhaps use rustdoc-like syntax to ignore fewer tests? We'd want to render with rustdoc and not link to GitHub in that case, but it'd be nice to have more than just a handful of doc blocks un-ignored :)

**Comment by ahmedcharles** [comment]
I'm not familiar enough with rustdoc syntax to do what you're suggesting. Any specific suggestions (or pointers to documentation) would be helpful.

The CI failures are because I spelt tutorial wrong. This should work.

**Comment by alexcrichton** [comment]
Oh sure, I just meant along the lines of:

    ```rust
    # let x = 3;
    println!("{}", x);
    ```

That way the rendered version online doesn't have the code that starts with `#` but everything is a testable example.

**Comment by ahmedcharles** [comment]
I was looking for how to do that! Thanks.

File: tutorial/TUTORIAL.md [hunk 1]
```diff
@@ -841,20 +1049,27 @@ into effect.
...
+# use futures::{Future, Oneshot, oneshot, Poll};
 struct MyFuture {
     inner: Oneshot<i32>,
 }
 
 fn foo() -> MyFuture {
     let (tx, rx) = oneshot();
     // ...
-    MyFuture { inner: tx }
+    MyFuture { inner: rx }
```
Comment (ahmedcharles) [inline tutorial/TUTORIAL.md]: I think this was bug in the tutorial before. I guess `rustdoc` tests are awesome.

**Comment by ahmedcharles** [comment]
Thanks for the enthusiastic response to the PR, it feels good to have work be appreciated. :)

Is there anything else I need to make this work well?

**Comment by alexcrichton** [comment]
Awesome, thanks! Looks like there's a merge conflict now with master, and yeah let's get deployment working here as well.

As part of the `after_success` step you'll want to run `rustdoc --test` and emit it into `target/doc` I think. That way it should get uploaded to `alexcrichton.com/futures-rs/TUTORIAL.html` I believe.

**Comment by ahmedcharles** [comment]
@alexcrichton It should work now, I think.

**Comment by alexcrichton** [comment]
Looks like travis failures might be legitimate?

**Comment by ahmedcharles** [comment]
Locally, I get:

```
$ cargo build --manifest-path tutorial/Cargo.toml
   Compiling cfg-if v0.1.0
   Compiling pkg-config v0.3.8
   Compiling bitflags v0.7.0
   Compiling openssl v0.8.2
   Compiling libc v0.2.15
   Compiling net2 v0.2.26
   Compiling void v1.0.2
   Compiling semver v0.1.20
   Compiling rustc_version v0.1.7
   Compiling lazy_static v0.2.1
   Compiling openssl-sys v0.7.17
   Compiling lazycell v0.4.0
   Compiling slab v0.3.0
   Compiling openssl-verify v0.2.0
   Compiling log v0.3.6
   Compiling futures v0.1.0 (https://github.com/alexcrichton/futures-rs#7172f735)
   Compiling scoped-tls v0.1.0
   Compiling bitflags v0.4.0
   Compiling nix v0.6.0
   Compiling mio v0.6.0
   Compiling tokio-core v0.1.0 (https://github.com/tokio-rs/tokio-core#83784fd9)
   Compiling tokio-tls v0.1.0 (https://github.com/tokio-rs/tokio-tls#21862697)
   Compiling tutorial v0.1.0 (file:///home/acharles/rust/code/futures-rs/tutorial)
$ rustdoc -L tutorial/target/debug -L tutorial/target/debug/deps --test tutorial/TUTORIAL.md

running 29 tests
test Getting_started_with_futures_0 ... ok
test Getting_started_with_futures_1 ... ok
test Getting_started_with_futures_10 ... ok
test Getting_started_with_futures_11 ... ok
test Getting_started_with_futures_12 ... ok
test Getting_started_with_futures_13 ... ok
test Getting_started_with_futures_14 ... ok
test Getting_started_with_futures_15 ... ok
test Getting_started_with_futures_16 ... ok
test Getting_started_with_futures_17 ... ok
test Getting_started_with_futures_18 ... ok
test Getting_started_with_futures_19 ... ok
test Getting_started_with_futures_2 ... ok
test Getting_started_with_futures_20 ... ok
test Getting_started_with_futures_21 ... ok
test Getting_started_with_futures_22 ... ok
test Getting_started_with_futures_23 ... ok
test Getting_started_with_futures_24 ... ok
test Getting_started_with_futures_25 ... ok
test Getting_started_with_futures_26 ... ok
test Getting_started_with_futures_27 ... ok
test Getting_started_with_futures_3 ... ok
test Getting_started_with_futures_4 ... ok
test Getting_started_with_futures_5 ... ok
test Getting_started_with_futures_6 ... ok
test Getting_started_with_futures_7 ... ok
test Getting_started_with_futures_8 ... ok
test Getting_started_with_futures_9 ... ok
test This_was_tested_on_nightly__but_is_ignored_due_to__code_futures__code__running_on_stable__0 ... ignored

test result: ok. 28 passed; 0 failed; 1 ignored; 0 measured

$
```

Any ideas why it doesn't work on travis?

**Comment by alexcrichton** [comment]
Maybe a different build directory structure or perhaps a stale directory?

**Comment by alexcrichton** [comment]
We've now removed the tutorial in favor of https://tokio.rs/, so I'm going to close this.

**Pull request closed without merging**